lib/_tls_wrap.js

```diff
@@ -989,7 +989,10 @@ exports.connect = function(/* [port, host], options, cb */) {
     var verifyError = socket._handle.verifyError();
 
     // Verify that server's identity matches it's certificate's names
-    if (!verifyError) {
+    // Unless server has resumed our existing session
+    if (!verifyError &&
+        (!options.session ||
+         Buffer.compare(options.session, socket.getSession()) !== 0)) {
```

Comment:
After some consideration - I don't think that it might be the case at this point in runtime. But I will add guard just in case.